src/validation.cpp

```diff
@@ -4982,6 +4991,91 @@ bool ChainstateManager::PopulateAndValidateSnapshot(
     return true;
 }
 
+bool ChainstateManager::completeSnapshotValidation(CChainState& validation_chainstate)
+{
+    AssertLockHeld(cs_main);
+    assert(&validation_chainstate == m_ibd_chainstate.get());
+
+    CCoinsViewDB& ibd_coins_db = validation_chainstate.CoinsDB();
+    validation_chainstate.ForceFlushStateToDisk();
+
+    CCoinsStats ibd_stats{CoinStatsHashType::HASH_SERIALIZED};
+    auto breakpoint_fnc = [] { /* TODO insert breakpoint here? */ };
+
+    if (!GetUTXOStats(&ibd_coins_db, WITH_LOCK(::cs_main, return std::ref(m_blockman)), ibd_stats, breakpoint_fnc)) {
+        LogPrintf("[snapshot] failed to generate stats for validation coins db\n");
+        return false;
```

Comment:
Should probably de-activate the existing snapshot chainstate as in the `if (snapshot_invalid)` conditional.